11-functions.Rmd

```diff
@@ -8,3 +8,117 @@ ottrpal::set_knitr_image_path()
 ```{r, fig.align='center', out.width="100%", echo = FALSE, fig.alt= "Major point!! example image"}
 ottrpal::include_slide("https://docs.google.com/presentation/d/1MNHf8JpolaEP_vQ_kB-1xRBF9wo3haCArRu117hBoHA/edit#slide=id.g21a84b32106_0_53")
 ```
+
+### Why write a custom function?
+
+There's a lot of reasons writing your own custom functions might be a thing you need to do.
+
+#### Cutting down on repetitive code
+
+We've discussed how DRY code is easier to maintain and understand! It is easier to fix code in one place as opposed to three or more places. A general guideline is that if you need two use a same chunk of code more than twice you probably should invest the time to make a custom function for it.
+
+#### More readable code
+
+Custom functions can allow you to organize code into manageable chunks.
+By creating custom functions it can be easier to organize our code in a way thats more readable.
+
+
+### Writing a function
+
+A common way that a custom function might come about is:
+
+1. You've written some initial code that takes some object as input.
+2. You now realize you will need to run this series of code more than twice.
+
+To get started you can follow these steps to make a custom function:
+
+### Starting a new function
+
+Open up a new R script file and copy and paste our basic template we have below.
+Your basic custom function is going to follow this type of structure:
+
+We will walk through how to use this template here.
+
+This top part of the template is the documentation. It's formatted in a way that would
+allow you to incorporate it into a package if you would like.
+
+1. It's likely you might already have some code drafted that has some of the code you need for your function.
+paste this code in the part of this template that says `# Your main code here`.
+
+```
```

Comment:
could we move the template up to right after we say open a new script

Reply:
Done!